test/e2e/storage/csimock/csi_node_volume_stats.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csimock
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/onsi/gomega"
+	"google.golang.org/grpc/codes"
+
+	csipbv1 "github.com/container-storage-interface/spec/lib/go/csi"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/features"
+	kubeletmetrics "k8s.io/kubernetes/pkg/kubelet/metrics"
+	"k8s.io/kubernetes/test/e2e/feature"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2emetrics "k8s.io/kubernetes/test/e2e/framework/metrics"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	e2epv "k8s.io/kubernetes/test/e2e/framework/pv"
+	"k8s.io/kubernetes/test/e2e/storage/drivers"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+)
+
+var _ = utils.SIGDescribe("CSI Mock Node Volume Stats", feature.CSIVolumeHealth, framework.WithFeatureGate(features.CSIVolumeHealth), func() {
+	f := framework.NewDefaultFramework("csi-mock-node-volume-stats")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+	m := newMockDriverSetup(f)
+
+	f.Context("CSI Mock Node Volume Stats", f.WithSlow(), func() {
+		trackedCalls := []string{
+			"NodeGetVolumeStats",
+		}
+		tests := []struct {
+			name                        string
+			expectedCalls               []csiCall
+			nodeVolumeStatRequired      bool
+			nodeVolumeConditionRequired bool
+			nodeAbnormalVolumeCondition bool
+		}{
+			{
+				name:                        "volume stats not returned",
+				expectedCalls:               []csiCall{},
+				nodeVolumeStatRequired:      false,
```

Comment:
Remove `nodeVolumeStatRequired` since we are not testing volume stats here really.